packages/block-library/src/navigation/index.php

```diff
@@ -278,29 +392,33 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 
 	$inner_blocks = $block->inner_blocks;
 
-	// If `__unstableLocation` is defined and:
+	// Ensure that blocks saved with the legacy ref attribute name (navigationMenuId) continue to render.
+	if ( array_key_exists( 'navigationMenuId', $attributes ) ) {
+		$attributes['ref'] = $attributes['navigationMenuId'];
+	}
+
+	// If:
+	// - the gutenberg plugin is active
+	// - `__unstableLocation` is defined
 	// - we have menu items at the defined location
-	// - we don't have a relationship to a `wp_navigation` Post (via `navigationMenuId`).
+	// - we don't have a relationship to a `wp_navigation` Post (via `ref`).
 	// ...then create inner blocks from the classic menu assigned to that location.
 	if (
+		defined( 'IS_GUTENBERG_PLUGIN' ) &&
```

Comment:
```suggestion
		defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN &&
```

`defined()` tells us that the constant exists in memory, but doesn't tell us if it's set to `true`. Theoretically, defined should only happen in the plugin and thus be set to `true` by default. Checking the value of the constant is both future-proof and guards against unknowns such as that constant being defined elsewhere.